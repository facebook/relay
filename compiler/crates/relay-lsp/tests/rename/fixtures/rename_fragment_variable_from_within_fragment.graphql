query MeQuery($include_firstname: Boolean!) {
  me {
    ...UserFragment(return_firstname: $include_firstname)
  }
}

fragment UserFragment($return_firstname: Boolean!) on User {
  name @skip(if: $return_firstname)
  ... @defer(if: $return_fi|rstname) {
    og_name: name
  }
  firstName(if: $return_firstname)
}

fragment CommentFragment on Comment {
  author {
    ...UserFragment(return_firstname: true)
  }
}