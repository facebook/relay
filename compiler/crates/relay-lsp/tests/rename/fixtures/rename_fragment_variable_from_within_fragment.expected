==================================== INPUT ====================================
query MeQuery($include_firstname: Boolean!) {
  me {
    ...UserFragment(return_firstname: $include_firstname)
  }
}

fragment UserFragment($return_firstname: Boolean!) on User {
  name @skip(if: $return_firstname)
  ... @defer(if: $return_fi|rstname) {
    og_name: name
  }
  firstName(if: $return_firstname)
}

fragment CommentFragment on Comment {
  author {
    ...UserFragment(return_firstname: true)
  }
}
==================================== OUTPUT ===================================
query MeQuery($include_firstname: Boolean!) {
  me {
    ...UserFragment(RENAMED: $include_firstname)
  }
}

fragment UserFragment($RENAMED: Boolean!) on User {
  name @skip(if: $RENAMED)
  ... @defer(if: $RENAMED) {
    og_name: name
  }
  firstName(if: $RENAMED)
}

fragment CommentFragment on Comment {
  author {
    ...UserFragment(RENAMED: true)
  }
}
